new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java

```diff
@@ -222,8 +206,288 @@ public void testServerPodsRestartByChangingResource() {
+    podsWithTimeStamps = getPodsWithTimeStamps();
+
+    //print out the original IncludeServerOutInPodLog
+    Boolean includeServerOutInPodLog = domain1.getSpec().getIncludeServerOutInPodLog();
+    logger.info("Original IncludeServerOutInPodLog is: {0}", includeServerOutInPodLog);
+
+    //change includeServerOutInPodLog: true --> includeServerOutInPodLog: false
+    StringBuffer patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/includeServerOutInPodLog\",")
+        .append("\"value\": ")
+        .append(false)
+        .append("}]");
+    logger.info("PatchStr for includeServerOutInPodLog: {0}", patchStr.toString());
+
+    boolean cmPatched = patchDomainResource(domainUid, domainNamespace, patchStr);
+    assertTrue(cmPatched, "patchDomainCustomResource(IncludeServerOutInPodLog) failed");
+
+    domain1 = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+        String.format("getDomainCustomResource failed with ApiException when tried to get domain %s in namespace %s",
+            domainUid, domainNamespace));
+    assertNotNull(domain1, domain1 + " is null");
+    assertNotNull(domain1.getSpec(), domain1 + "/spec is null");
+
+    includeServerOutInPodLog = domain1.getSpec().getIncludeServerOutInPodLog();
+    logger.info("In the new patched domain IncludeServerOutInPodLog is: {0}",
+        includeServerOutInPodLog);
+    assertFalse(includeServerOutInPodLog, "IncludeServerOutInPodLog was not updated");
+
+    // verify the server pods are rolling restarted and back to ready state
+    logger.info("Verifying rolling restart occurred for domain {0} in namespace {1}",
+        domainUid, domainNamespace);
+    assertTrue(assertDoesNotThrow(
+        () -> verifyRollingRestartOccurred(podsWithTimeStamps, 1, domainNamespace),
+        "More than one pod was restarted at same time"),
+        String.format("Rolling restart failed for domain %s in namespace %s", domainUid, domainNamespace));
+
+  }
+
+  /**
+   * Modify domain scope serverPod env property on the domain resource.
+   * Verify all pods are restarted and back to ready state.
+   * The env property tested: "-Dweblogic.StdoutDebugEnabled=false" --> "-Dweblogic.StdoutDebugEnabled=true".
+   */
+  @Test
+  @DisplayName("Verify server pods are restarted by changing serverPod env property")
+  @Slow
+  public void testServerPodsRestartByChangingEnvProperty() {
+    // get the original domain resource before update
+    Domain domain1 = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace),
+        String.format("getDomainCustomResource failed with ApiException when tried to get domain %s in namespace %s",
+            domainUid, domainNamespace));
+    assertNotNull(domain1, domain1 + " is null");
+    assertNotNull(domain1.getSpec(), domain1 + " /spec/serverPod is null");
```

Comment:
If the domain1 is Null, it will throw a NPE. We can change the statement to
assertNotNull(domain1, "Got Null Domain Resource after patching ");

Reply:
Done